src/main/java/com/hubspot/jinjava/lib/filter/time/DateTimeFormatHelper.java

```diff
@@ -0,0 +1,87 @@
+package com.hubspot.jinjava.lib.filter.time;
+
+import com.hubspot.jinjava.interpret.InvalidArgumentException;
+import com.hubspot.jinjava.interpret.JinjavaInterpreter;
+import com.hubspot.jinjava.lib.fn.Functions;
+import com.hubspot.jinjava.objects.date.InvalidDateFormatException;
+import java.time.DateTimeException;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.time.format.FormatStyle;
+import java.util.IllformedLocaleException;
+import java.util.Locale;
+import java.util.Optional;
+import java.util.function.Function;
+
+final class DateTimeFormatHelper {
+  private final String name;
+  private final Function<FormatStyle, DateTimeFormatter> cannedFormatterFunction;
+
+  DateTimeFormatHelper(
+    String name,
+    Function<FormatStyle, DateTimeFormatter> cannedFormatterFunction
+  ) {
+    this.name = name;
+    this.cannedFormatterFunction = cannedFormatterFunction;
+  }
+
+  String format(Object var, JinjavaInterpreter interpreter, String... args) {
+    String format = arg(args, 0).orElse("medium");
+    ZoneId zoneId = arg(args, 1).map(this::parseZone).orElse(ZoneOffset.UTC);
+    Locale locale = arg(args, 2)
+      .map(this::parseLocale)
+      .orElseGet(() -> interpreter.getConfig().getLocale());
+
+    return buildFormatter(format)
+      .withLocale(locale)
+      .format(Functions.getDateTimeArg(var, zoneId));
```

Comment:
@tkindy How catastrophic do you think it would be to stop defaulting to the current datetime if `var` is null for these new datetime filters? This would eliminate an introduction of a deferred value and allow us to make a clean transition off of the `datetimeformat` filter.

Reply:
Sorry, just seeing this.

I think that'd be fine, I can't imagine anyone's using the default time here. Want me to make that change?